### PR TITLE
Replace boltdb with fast-store

### PR DESCRIPTION
**Pull request by vcastellm**
Replacing BoltDB with fast-store increased the speed of the Raft log operations enormously, allowing the user to choose the degree of duration in disk.

Under my tests, with the same degree of duration as boltdb (High) the speed is x2 but using the Mid duration, the default that sync to disk every second, the speed increases tenfold.

Considering that Dkron is not the source of truth for the data it stores, the cost-benefit of using less durability is high.

Note: This will be included in the upcoming v4.x

### CODE REVIEW

File: dkron/agent.go [hunk 1]
```diff
@@ -312,7 +312,7 @@ func (a *Agent) setupRaft() error {
 		}
 
 		// Create the BoltDB backend
```
Comment (yvanoers) [inline dkron/agent.go:314]: You missed updating the comment here.

**Review by yvanoers — CHANGES_REQUESTED** [review]
Besides the comment, LGTM.
But does this change require users to do some kind of migration, or will this work transparently?

**Comment by victorcoder** [comment]
Yes @yvanoers this is a breaking change, this is why this will be published in 4.x, you can not upgrade in place but it only requires of a rolling deploy.

**Review by yvanoers — APPROVED** [review]
LGTM!

**Comment by yvanoers** [comment]
> Yes @yvanoers this is a breaking change, this is why this will be published in 4.x, you can not upgrade in place but it only requires of a rolling deploy.

Ok; is there, or will there be documentation on how to do that? I'm unaware of any.

**Comment by victorcoder** [comment]
@yvanoers doc for upgrade strategies here https://dkron.io/usage/upgrade/#rolling-upgrade

**Comment by yvanoers** [comment]
Ah yes, great!
And here's a PR to improve some of the text: #980 :)

**Comment by vcastellm** [comment]
Superseded by #1172

**Pull request closed without merging**